Source/WebCore/dom/WindowEventLoop.cpp

```diff
@@ -147,13 +148,14 @@ void WindowEventLoop::opportunisticallyRunIdleCallbacks()
     });
 }
 
-bool WindowEventLoop::shouldEndIdlePeriod()
+bool WindowEventLoop::shouldEndIdlePeriod(MonotonicTime now)
 {
     if (hasTasksForFullyActiveDocument())
         return true;
     if (!microtaskQueue().isEmpty())
         return true;
-    if (!m_pagesWithRenderingOpportunity.isEmptyIgnoringNullReferences())
+    auto renderingTime = nextRenderingTime();
+    if (renderingTime && *renderingTime < now + IdleCallbackDurationExpectation)
```

Comment:
The existing similar test involving IdleCallbackDurationExpectation uses `<=`

```
    if (timerTime && *timerTime <= now + IdleCallbackDurationExpectation) {
```

Probably not significant but I was wondering why this is different here?

Reply:
I guess we can use <= here as well. Will fix.

Reply:
Actually, we seem to use `<` more often than not so let me fix the other site in a follow up.